src/signin/scraper.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+const puppeteer = require("puppeteer");
+const url = require("url");
+const querystring = require("querystring");
+
+async function startScraper(tokenHref, headless = false) {
+  const originHost = url.parse(tokenHref).host;
+  const browser = await puppeteer.launch({headless});
+  const page = await browser.newPage();
+
+  await page.setRequestInterception(true);
+  return {
+    originHost,
+    browser,
+    page,
+  };
+}
+
+async function submitSecret({page, secret, timeout, reject}) {
+  try {
+    await page.click('button[title="log in with e-mail"]');
```

Comment:
This seems very prone to failure. If any of these fields in the html page change this will fail.

Reply:
I agree. If you have some ideas feel free to change. If Toby changes the layout I will change the code accordingly.